modules/auxiliary/gather/qnap_lfi.rb

```diff
@@ -0,0 +1,205 @@
+require 'metasploit/framework/hashes/identify'
+
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'QNAP QTS and PhotoStation Local File Inclusion',
+        'Description' => %q{
+          This module abuses a vulnerability in QNAP QTS and PhotoStation that allows an
+          unauthenticated user to download files off the file system, and because the server
+          runs as root, it's possible to include sensitive files, including ssh private keys and
+          password hashes.
+          This module has been tested with QNAP QTS 4.3.3
+        },
+        'Author' =>
+          [
+            'Henry Huang', # Vulnerability discovery
+            'Redouane NIBOUCHA <rniboucha[at]yahoo.fr>' # MSF module
+          ],
+        'License' => MSF_LICENSE,
+        'Platform' => %w[linux],
+        'References' =>
+          [
+            [ 'CVE', '2019-7192'],
+            [ 'CVE', '2019-7193'],
+            [ 'CVE', '2019-7194'],
```

Comment:
i think this was the one that wasn't applicable as it was the file write of the php code